internal/controller/fluxinstance_artifact_controller.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/fluxcd/pkg/runtime/patch"
+	corev1 "k8s.io/api/core/v1"
+	kuberecorder "k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fluxcdv1 "github.com/controlplaneio-fluxcd/flux-operator/api/v1"
+	"github.com/controlplaneio-fluxcd/flux-operator/internal/builder"
+)
+
+// FluxInstanceArtifactReconciler reconciles the distribution artifact of a FluxInstance object
+type FluxInstanceArtifactReconciler struct {
+	client.Client
+	kuberecorder.EventRecorder
+
+	StatusManager string
+}
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *FluxInstanceArtifactReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, retErr error) {
+	obj := &fluxcdv1.FluxInstance{}
+	if err := r.Get(ctx, req.NamespacedName, obj); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Skip reconciliation if the object is under deletion.
+	if !obj.ObjectMeta.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	// Skip reconciliation if the object has the reconcile annotation set to 'disabled'.
+	if obj.IsDisabled() {
+		return ctrl.Result{}, nil
+	}
+
+	// Skip reconciliation if the object does not have a distribution artifact specified.
+	if obj.Spec.Distribution.Artifact == "" {
+		return ctrl.Result{}, nil
+	}
+
```

Comment:
With this we avoid a race condition when the FluxInstance is first created, we don't want this reconciler to run then.
```suggestion
if obj.Status == nil || obj.Status.LastArtifactRevision == "" {
return requeueArtifactAfter(obj *fluxcdv1.FluxInstance), nil
}
```

Reply:
fixed, thanks! (btw `obj.Status` is not a pointer)